homeassistant/components/homee/cover.py

```diff
@@ -0,0 +1,262 @@
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeeConfigEntry, helpers
+from .entity import HomeeNodeEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+OPEN_CLOSE_ATTRIBUTES = [
+    AttributeType.OPEN_CLOSE,
+    AttributeType.SLAT_ROTATION_IMPULSE,
+    AttributeType.UP_DOWN,
+]
+POSITION_ATTRIBUTES = [AttributeType.POSITION, AttributeType.SHUTTER_SLAT_POSITION]
+
+
+def get_cover_features(
+    node: HomeeNodeEntity,
+) -> tuple[CoverEntityFeature, AttributeType]:
+    """Determine the supported cover features of a homee node based on the available attributes."""
+    features = CoverEntityFeature(0)
+    open_close = None
+
+    # We assume, that no device has UP_DOWN and OPEN_CLOSE, but only one of them.
+    if node.has_attribute(AttributeType.UP_DOWN) or node.has_attribute(
+        AttributeType.OPEN_CLOSE
+    ):
+        if node.has_attribute(AttributeType.UP_DOWN):
+            open_close = AttributeType.UP_DOWN
+        else:
+            open_close = AttributeType.OPEN_CLOSE
+
+        if node.get_attribute(open_close).editable:
+            features |= (
+                CoverEntityFeature.OPEN
+                | CoverEntityFeature.CLOSE
+                | CoverEntityFeature.STOP
+            )
+
+    # Check for up/down position settable.
+    if node.has_attribute(AttributeType.POSITION):
+        if node.get_attribute(AttributeType.POSITION).editable:
+            features |= CoverEntityFeature.SET_POSITION
+
+    if node.has_attribute(AttributeType.SLAT_ROTATION_IMPULSE):
+        features |= CoverEntityFeature.OPEN_TILT | CoverEntityFeature.CLOSE_TILT
+
+    if node.has_attribute(AttributeType.SHUTTER_SLAT_POSITION):
+        features |= CoverEntityFeature.SET_TILT_POSITION
+
+    return features, open_close
+
+
+def get_device_class(node: HomeeNode) -> CoverDeviceClass | None:
+    """Determine the device class a homee node based on the node profile."""
+    if node.profile == NodeProfile.GARAGE_DOOR_OPERATOR:
+        return CoverDeviceClass.GARAGE
+
+    if node.profile == NodeProfile.SHUTTER_POSITION_SWITCH:
+        return CoverDeviceClass.SHUTTER
+
+    return None
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: HomeeConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Add the homee platform for the cover integration."""
+
+    devices: list[HomeeCover] = []
+    nodes = helpers.get_imported_nodes(config_entry)
+    devices.extend(
+        HomeeCover(node, config_entry) for node in nodes if is_cover_node(node)
+    )
+
+    if devices:
+        async_add_devices(devices)
+
+
+def is_cover_node(node: HomeeNode) -> bool:
+    """Determine if a node is controllable as a homee cover based on its profile and attributes."""
+    return node.profile in [
+        NodeProfile.ELECTRIC_MOTOR_METERING_SWITCH,
+        NodeProfile.ELECTRIC_MOTOR_METERING_SWITCH_WITHOUT_SLAT_POSITION,
```

Comment:
What are those 2?

Reply:
The node profiles are the types of devices in Homee [see here](https://github.com/Taraman17/pyHomee/blob/d1c3e397a61abadf6fa5d6e9f37923460b8db833/pyHomee/const.py#L6) and these are the profiles known to be covers. It might be, that this list grows in the future, since the API is not well documented and it's a lot reverse engineering and trial and error.

Reply:
Oh I get that, but what is an electric motor metering switch?

Reply:
It's a switchable Motor with a meter for W and kWh.

Reply:
I guess. All that node profiles are taken from an unofficial document on the Homee API.